### DISCUSSION TITLE
Preload Live streams

### DISCUSSION THREAD

**Discussion by Usama99786** [pedroSG94/RootEncoder#1516] [Q&A]
I have a list of livestreams that I play in a recyclerview where the user can scroll on stream at a time like it is done in tiktok. Right now, I have implemented it using webRtc, but the problem is that there is a loading time of 2-3 seconds for each stream. I want to know if I use this library can I preload the livestreams using exoplayer? So when i play the stream it is played instantly like tiktok

**Comment by pedroSG94** [comment]
Hello,

Currently exoplayer can reproduce rtmp and rtsp stream so you can use this library to stream to a media server in that protocols and reproduce the stream using exoplayer.

**Reply by Usama99786** [reply]
Hi @pedroSG94 thanks for replying, I know i can play these streams on exoplayer, My query is Can i preload some chunks of the streams beforehand, So when I play the streams, There is no loading time and it is played instantly?

**Reply by pedroSG94** [reply]
I don't know if you can preload a stream in exoplayer.
That is a question that you should ask in the exoplayer repository
